org.envirocar.app/src/org/envirocar/app/views/tracklist/TrackListLocalCardFragment.java

```diff
@@ -167,9 +167,14 @@ public void onUploadTrackClicked(Track track) {
             @Override
             public void onExportTrackClicked(Track track) {
                 LOG.info(String.format("onExportTrackClicked(%s)", track.getTrackID()));
-                track.updateMetadata(new TrackMetadata(Util.getVersionString(getActivity()),
-                        mUserManager.getUser().getTermsOfUseVersion()));
-                exportTrack(track);
+                if  (mUserManager.getUser()!=null) {
+                    track.updateMetadata(new TrackMetadata(Util.getVersionString(getActivity()),
+                            mUserManager.getUser().getTermsOfUseVersion()));
+                    exportTrack(track);
+                }
+                else  {
+                    showSnackbar(R.string.trackview_cannot_export_track);
```

Comment:
I guess you can simply set the terms of use version to null if no user exists. It is okay when not logged in users can export a track.